pkg/handlers/license.go

```diff
@@ -338,26 +338,35 @@ func (h *Handler) UploadNewLicense(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		logger.Error(err)
 		uploadLicenseResponse.Error = err.Error()
-		JSON(w, 500, uploadLicenseResponse)
+		JSON(w, http.StatusInternalServerError, uploadLicenseResponse)
 		return
 	}
 
 	if !verifiedLicense.Spec.IsAirgapSupported {
 		// complete the install online
 		createAppOpts := online.CreateOnlineAppOpts{
 			PendingApp: &installationtypes.PendingApp{
-				ID:          a.ID,
-				Slug:        a.Slug,
-				Name:        a.Name,
-				LicenseData: uploadLicenseRequest.LicenseData,
+				ID:           a.ID,
+				Slug:         a.Slug,
+				Name:         a.Name,
+				LicenseData:  uploadLicenseRequest.LicenseData,
+				VersionLabel: installationParams.AppVersionLabel,
 			},
 			UpstreamURI: upstreamURI,
 		}
 		kotsKinds, err := online.CreateAppFromOnline(createAppOpts)
 		if err != nil {
 			logger.Error(err)
 			uploadLicenseResponse.Error = err.Error()
-			JSON(w, 500, uploadLicenseResponse)
+			JSON(w, http.StatusInternalServerError, uploadLicenseResponse)
+			return
+		}
+
+		err = kotsutil.RemoveAppVersionLabelFromInstallationParams(kotsadmtypes.KotsadmConfigMap)
```

Comment:
why not in `CreateAppFromOnline`? since that's called from multiple places, like resuming an online install.